ddtrace/sampler.py

```diff
@@ -150,32 +149,29 @@ def set_sample_rate(
         # type: (...) -> None
         self._by_service_samplers[self._key(service, env)] = RateSampler(sample_rate)
 
-    def _set_priority(self, span, priority):
-        # type: (Span, int) -> None
-        span.context.sampling_priority = priority
-        span.sampled = priority > 0  # Positive priorities mean it was kept
-
-    def _set_sampler_decision(self, span, sampler, sampled):
-        # type: (Span, RateSampler, bool) -> None
-        priority = AUTO_KEEP if sampled else AUTO_REJECT
-        self._set_priority(span, priority)
-        if sampler is self._default_sampler:
-            mechanism = SamplingMechanism.DEFAULT
-        else:
-            mechanism = SamplingMechanism.AGENT_RATE
-            span.set_metric(SAMPLING_AGENT_DECISION, sampler.sample_rate)
-        set_sampling_decision_maker(span.context, mechanism)
+    def sample(self, span, allow_false=False):
+        sampled, sampler = self._make_sampling_decision(span)
+        _set_sampling_tags(
+            span,
+            sampled,
+            sampler.sample_rate,
+            self._choose_priority_category(sampler is self._default_sampler),
+        )
+        return not allow_false or sampled
+
+    def _choose_priority_category(self, default_was_used):
+        # type: (bool) -> str
+        if default_was_used:
+            return _PRIORITY_CATEGORY.DEFAULT
+        return _PRIORITY_CATEGORY.AUTO
 
-    def sample(self, span):
-        # type: (Span) -> bool
+    def _make_sampling_decision(self, span):
+        # type: (Span) -> Tuple[bool, BaseSampler]
         env = span.get_tag(ENV_KEY)
         key = self._key(span.service, env)
-
         sampler = self._by_service_samplers.get(key) or self._default_sampler
```

Comment:
Since it's part of the public API, should we change `allow_false = True`  to the default so `sample` would return the results of `_make_sampling_decision()`?